mvx-esdt-safe/src/briding_mechanism.rs

```diff
@@ -0,0 +1,60 @@
+use error_messages::{
+    MINT_AND_BURN_ROLES_NOT_FOUND, TOKEN_ID_IS_NOT_TRUSTED, TOKEN_IS_FROM_SOVEREIGN,
+};
+use multiversx_sc::imports::*;
+
+pub const TRUSTED_TOKEN_IDS: [&str; 1] = ["USDC-c76f1f"];
+
+#[multiversx_sc::module]
+pub trait BridgingMechanism:
+    cross_chain::storage::CrossChainStorage + multiversx_sc_modules::only_admin::OnlyAdminModule
+{
+    #[only_admin]
+    #[endpoint(setTokenBurnMechanism)]
+    fn set_token_burn_mechanism(&self, token_id: TokenIdentifier) {
+        let token_esdt_roles = self.blockchain().get_esdt_local_roles(&token_id);
+
+        require!(
+            token_esdt_roles.contains(EsdtLocalRoleFlags::MINT)
+                && token_esdt_roles.contains(EsdtLocalRoleFlags::BURN),
+            MINT_AND_BURN_ROLES_NOT_FOUND
+        );
+
+        require!(
+            TRUSTED_TOKEN_IDS
+                .iter()
+                .any(|trusted_token_id| TokenIdentifier::from(*trusted_token_id) == token_id),
+            TOKEN_ID_IS_NOT_TRUSTED
```

Comment:
For further references, try to end this kind of variable with _ERR_MSG like `TOKEN_ID_IS_NOT_TRUSTED_ERR_MSG`.